php/libraries/Candidate.class.inc

```diff
@@ -613,23 +613,20 @@ class Candidate
     /**
      * Returns first visit for a candidate based on Date_visit
      *
-     * @return string The visit label for the candidate's first visit
+     * @return string|null The visit label for the candidate's first visit
      */
     public function getFirstVisit()
     {
         $factory = NDB_Factory::singleton();
         $db      = $factory->database();
 
         $candID = $this->getCandID();
-
-        $query  = "SELECT Visit_label
-            FROM session s JOIN candidate c ON (c.CandID = s.CandID)
-            WHERE c.CandID=:cid AND Date_Visit IS NOT NULL
-            ORDER BY Date_Visit";
+        $query  = "SELECT Visit_label FROM session 
+          WHERE CandID = :cid AND Date_visit IS NOT NULL 
+          ORDER BY Date_visit, ID ASC LIMIT 1";
         $where  = array('cid' => $candID);
-        $vLabel = $db->pselectOne($query, $where);
-
-        return $vLabel;
+        $result = $db->pselectCol($query, $where);
+        return array_shift($result);
```

Comment:
This is failing Travis when the value of `$result` is null because `array_shift` can only operate on `arrays`. 

I think you just need to do this:

`return is_array($result)? array_shift($result) : null;`

Reply:
I went to look up the documentation to see if `array_shift($result ?? []);` would work too and it says it "Returns the shifted value, or NULL if array is empty or is not an array. "... so given that the documentation (seems to be?) wrong about the not an array bit, it's probably safer to do your check than rely on the rest of the documentation being correct. (I think PHPCS might insist on a space before the `?` though)

Reply:
$result in this case is also `null` so the `??` condition wouldn't go off. It only works for values that are not set, iirc.

Reply:
`??` is the null coalesce operator. It works for unset or null value (but not other cases that would be considered "empty" in PHP).